### PR TITLE
Added scopes, endpoint security and errors

### PR DESCRIPTION
**Pull request by nissimsan**
- Added scopes from https://github.com/w3c-ccg/traceability-interop/issues/124
- Assigned to endpoints
- Added 401 and 403 errors

### CODE REVIEW

File: docs/openapi/openapi.yml [hunk 1]
```diff
@@ -52,8 +52,14 @@ components:
           tokenUrl: https://example.com/oauth/token
           redirectUri: 'http://localhost:5000/oauth2-redirect.html'
           scopes:
-            'read:presentations': Grants permission to read submitted presentations
-            'write:presentations': Grants permission to submit presentations
+            'resolve:did': Grants permission to resolve DIDs
+            'create:vc': Grants permission issue Verifiable Credentials
+            'verify:vc': Grants permission verify Verifiable Credentials
+            'read:vc': Grants permission to get Verifiable Credentials
+            'update:vc': Grants permission to update the status of Verifiable Credentials
```
Comment (BenjaminMoe) [inline docs/openapi/openapi.yml:59]: Is this updating the vc or the revocation list?

Comment (nissimsan) [reply docs/openapi/openapi.yml:59]: It's updating the revocation list VC. So both, really. 
I added this myself as #124 didn't cover a scope for this (it's the only one I added). I'm definitely open to better naming or granularity. We could call it `revoke:vc`, but distinguishing between VC types (revocation list vs. all others) seems non-trivial?

**Review by mprorock — APPROVED** [review]

**Review by BenjaminMoe — CHANGES_REQUESTED** [review]
I think trying to be a cool kid and abbreviating `vc` and `vp` is going to hurt us in the long run. The main reason to abbreviate is to save space, we should focus on clarity. 

- `create:vc` -> `create:credential`
- `verify:vc` -> `verify:credential`
- `read:vc` -> `read:credential`
- `update:vc` -> `update:credential`
- `create:vp` -> `create:presentation`
- `verify:vp` -> `verify:presentation`
- `exchange:vp` -> `exchange:presentation`

File: docs/openapi/openapi.yml [hunk 2]
```diff
@@ -52,8 +52,14 @@ components:
           tokenUrl: https://example.com/oauth/token
           redirectUri: 'http://localhost:5000/oauth2-redirect.html'
           scopes:
-            'read:presentations': Grants permission to read submitted presentations
-            'write:presentations': Grants permission to submit presentations
+            'resolve:did': Grants permission to resolve DIDs
+            'create:vc': Grants permission issue Verifiable Credentials
```
Comment (BenjaminMoe) [inline docs/openapi/openapi.yml:56]:
```suggestion
           'create:verifiable_credential': Grants permission issue Verifiable Credentials
```

File: docs/openapi/openapi.yml [hunk 3]
```diff
@@ -52,8 +52,14 @@ components:
...
+            'verify:vc': Grants permission verify Verifiable Credentials
```
Comment (BenjaminMoe) [inline docs/openapi/openapi.yml:57]:
```suggestion
            'verify:verifiable_credential': Grants permission verify Verifiable Credentials
```

File: docs/openapi/openapi.yml [hunk 4]
```diff
@@ -52,8 +52,14 @@ components:
...
+            'read:vc': Grants permission to get Verifiable Credentials
```
Comment (BenjaminMoe) [inline docs/openapi/openapi.yml:58]:
```suggestion
            'read:verifiable_credential': Grants permission to get Verifiable Credentials
```

File: docs/openapi/openapi.yml (hunk 1 above)
Comment (BenjaminMoe) [inline docs/openapi/openapi.yml:59]:
```suggestion
            'update:verifiable_credential': Grants permission to update the status of Verifiable Credentials
```

File: docs/openapi/openapi.yml [hunk 5]
```diff
@@ -52,8 +52,14 @@ components:
...
+            'update:vc': Grants permission to update the status of Verifiable Credentials
+            'create:vp': Grants permission to proof Verifiable Presentations
```
Comment (BenjaminMoe) [inline docs/openapi/openapi.yml:60]:
```suggestion
            'create:verifiable_presentation': Grants permission to proof Verifiable Presentations
```

File: docs/openapi/openapi.yml [hunk 6]
```diff
@@ -52,8 +52,14 @@ components:
...
+            'verify:vp': Grants permission verify Verifiable Presentations
```
Comment (BenjaminMoe) [inline docs/openapi/openapi.yml:61]:
```suggestion
            'verify:verifiable_presentation': Grants permission verify Verifiable Presentations
```

File: docs/openapi/openapi.yml [hunk 7]
```diff
@@ -52,8 +52,14 @@ components:
...
+            'exchange:vp': Grants permission present Verifiable Presentations
```
Comment (BenjaminMoe) [inline docs/openapi/openapi.yml:62]:
```suggestion
            'exchange:verifiable_presentation': Grants permission present Verifiable Presentations
```

File: docs/openapi/resources/credential-issuer.yml [hunk 8]
```diff
@@ -4,6 +4,9 @@ post:
   description: Issue a credential
   tags:
     - Credentials
+  security:
+    - OAuth2:
+        - 'create:vc'
```
Comment (BenjaminMoe) [inline docs/openapi/resources/credential-issuer.yml:9]:
```suggestion
        - 'create:verifiable_credential'
```

File: docs/openapi/resources/credential-status.yml [hunk 9]
```diff
@@ -4,6 +4,9 @@ post:
   description: Updates the status of an issued credential.
   tags:
     - Credentials
+  security:
+    - OAuth2:
+        - 'update:vc'
```
Comment (BenjaminMoe) [inline docs/openapi/resources/credential-status.yml:9]:
```suggestion
       - 'update:verifiable_credential'
```

File: docs/openapi/resources/credential-verifier.yml [hunk 10]
```diff
@@ -4,6 +4,9 @@ post:
   description: Verify a credential
   tags:
     - Credentials
+  security:
+    - OAuth2:
+        - 'verify:vc'
```
Comment (BenjaminMoe) [inline docs/openapi/resources/credential-verifier.yml:9]:
```suggestion
        - 'verify:verifiable_credential'
```

File: docs/openapi/resources/credential.yml [hunk 11]
```diff
@@ -5,6 +5,9 @@ get:
   description: Get a verifiable credential by id. Required to make revocable credentials.
   tags:
     - Credentials
+  security:
+    - OAuth2:
+        - 'read:vc'
```
Comment (BenjaminMoe) [inline docs/openapi/resources/credential.yml:10]:
```suggestion
        - 'read:verifiable_credential'
```

File: docs/openapi/resources/presentation-prover.yml [hunk 12]
```diff
@@ -4,6 +4,9 @@ post:
   description: Create a presentation
   tags:
     - Presentations
+  security:
+    - OAuth2:
+        - 'create:vp'
```
Comment (BenjaminMoe) [inline docs/openapi/resources/presentation-prover.yml:9]:
```suggestion
        - 'create:verifiable_presentation'
```

File: docs/openapi/resources/presentation-verifier.yml [hunk 13]
```diff
@@ -4,6 +4,9 @@ post:
   description: Verify a presentation
   tags:
     - Presentations
+  security:
+    - OAuth2:
+        - 'verify:vp'
```
Comment (BenjaminMoe) [inline docs/openapi/resources/presentation-verifier.yml:9]:
```suggestion
        - 'verify:verifiable_presentation'
```

File: docs/openapi/resources/presentations.yml [hunk 14]
```diff
@@ -5,21 +5,23 @@ post:
     - Presentations
   security:
     - OAuth2:
-        - 'submit:presentations'
-
+        - 'exchange:vp'
```
Comment (BenjaminMoe) [inline docs/openapi/resources/presentations.yml:8]:
```suggestion
        - 'exchange:verifiable_presentation'
```

**Review by BenjaminMoe — APPROVED** [review]

**Review by OR13 — APPROVED** [review]

**Pull request merged**